### DISCUSSION TITLE
SetResult cannot be non-pointer.

### DISCUSSION THREAD

**Discussion by w570955342** [go-resty/resty#475] [Q&A] [answered]
When I run this：

func main() {

	result := make([]AutoGenerated, 0)
	res, err := resty.New().SetTimeout(time.Second*60).R().
		SetHeader("Content-Type", "application/json").
		// 注释说 Result object can be pointer or non-pointer.
		// 但是如果传切片，不是切片的指针，则不能获取数据。
		SetResult(result).  // len(result) 0
		//SetResult(&result).  // len(result) 100
		Get("http://air.htkjbjf.com:1337/releases")
	if err != nil {
		fmt.Printf("license req err, %s", err.Error())
	}
	if !res.IsSuccess() {
		fmt.Printf("license req fail, %+v", res)
	}
	//log.Println("result:", result)

	fmt.Println(len(res.Body()))
	fmt.Println(len(result))

}

case SetResult(result).  // len(result) 0
case SetResult(&result).  // len(result) 100

Can result be non-pointer?

https://github.com/go-resty/resty/blob/master/request.go#:~:text=Note%3A%20Result%20object%20can%20be%20pointer%20or%20non-pointer

**Comment by moorereason** [comment, marked as answer]
If you don't pass a pointer, you're passing a copy of the value to `SetResult`, so I think you would need to use `*Response.Result()` to retrieve the result:

```go
result = res.Result().(*[]AutoGenerated)
```

**Reply by w570955342** [reply]
I got it wrong before. Thank you very much.
